3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+*/
+
+#include "fastcv_hal_imgproc.hpp"
+#include "fastcv_hal_utils.hpp"
+#include <opencv2/core/base.hpp>
+#include <opencv2/core/utils/logger.hpp>
+#include <opencv2/imgproc.hpp>
+
+extern bool isInitialized;
+
+int fastcv_hal_medianBlur(
+    const uchar*    src_data,
+    size_t          src_step,
+    uchar*          dst_data,
+    size_t          dst_step,
+    int             width,
+    int             height,
+    int             depth,
+    int             cn,
+    int             ksize)
+{
+    // Do not support inplace case
+    if (src_data == dst_data)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("Inplace is not supported");
+
+    // The input image width and height should greater than kernel size
+    if ((height <= ksize) || (width <= ksize))
+        CV_HAL_RETURN_NOT_IMPLEMENTED("Input image size should be larger than kernel size");
+
+    // The input channel should be 1
+    if (cn != 1)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("Multi-channels is not supported");
+
+    INITIALIZATION_CHECK;
+
+    fcvStatus status;
+    int fcvFuncType = FCV_MAKETYPE(ksize,depth);
+
+    switch (fcvFuncType)
+    {
+        case FCV_MAKETYPE(3,CV_8U):
+        {
+            status = fcvFilterMedian3x3u8_v3(src_data, width, height, src_step, dst_data, dst_step,
+                fcvBorderType::FASTCV_BORDER_REPLICATE, 0);
+            break;
+        }
+        default:
+            CV_HAL_RETURN_NOT_IMPLEMENTED(cv::format("Ksize:%d, depth:%s is not supported", ksize, cv::depthToString(depth)));
+    }
+
+    CV_HAL_RETURN(status, hal_medianBlur);
+}
+
+int fastcv_hal_sobel(
+    const uchar*    src_data,
+    size_t          src_step,
+    uchar*          dst_data,
+    size_t          dst_step,
+    int             width,
+    int             height,
+    int             src_depth,
+    int             dst_depth,
+    int             cn,
+    int             margin_left,
+    int             margin_top,
+    int             margin_right,
+    int             margin_bottom,
+    int             dx,
+    int             dy,
+    int             ksize,
+    double          scale,
+    double          delta,
+    int             border_type)
+{
+
+    // @todo unused parameters
+    CV_UNUSED(scale);
+    CV_UNUSED(delta);
```

Comment:
Please return CV_HAL_RETURN_NOT_IMPLEMENTED, if the values are not default/expected. Otherwise user gets invalid result.